src/cached_object_store.rs

```diff
@@ -81,8 +121,19 @@ impl CachedObjectStore {
                         },
                     )
                     .await?;
+
                 let meta = result.meta.clone();
-                self.save_result(result).await.ok();
+                let mut ghost_guard = self.ghost.lock().await;
```

Comment:
i am not sure do i understand this expected behaviour correctly:

1. If the object is cached, return the cache.
2. When the object is accessed and is not contained in the ghost queue, put it into the ghost queue first, and fall back to the backend.
3. When the object is accessed and is already contained in the ghost queue, put it into the cache and remove it from the ghost queue.

if it is, i suppose there might have ways to encapsulate the details about ghost queue within some self-contained methods? that may allows the rest of the code to be less concerned with the details of the ghost queue, but focus more on the logic on the globe view.

Reply:
the ghost queue is expected to prevent the data from writing into disk, right? i guess it might possible to have something like this as its interface:

```
struct GhostQueue {
   ..
}

impl GhostQueue {
  fn need_cache(path: &Path) -> bool { }
  fn track_accessed(path: &Path);
}
```

Reply:
@KaguraMilet what do you think about this interface?

Reply:
This is not a problem, I can change it.

Reply:
I found that we still have `increase_ghost_capacity` in the cache side. Is this considered a detail that the caller must take care of?

Can you help explain a bit about the minimal interaction between the cache and ghost queue?